lib/CUDAKernels/src/CUDAKernels.jl

```diff
@@ -335,15 +293,15 @@ end
 # - private memory for each workitem
 ###
 
-@inline function Cassette.overdub(::CUDACtx, ::typeof(Scratchpad), ctx, ::Type{T}, ::Val{Dims}) where {T, Dims}
+@device_override @inline function Scratchpad(ctx, ::Type{T}, ::Val{Dims}) where {T, Dims}
     MArray{__size(Dims), T}(undef)
 end
 
 @inline function Cassette.overdub(::CUDACtx, ::typeof(__synchronize))
```

Comment:
```suggestion
@device_override @inline function __synchronize()
```

Seems to work in my tests